folly/CPortability.h

```diff
@@ -53,4 +53,725 @@
 # define FOLLY_DISABLE_ADDRESS_SANITIZER
 #endif
 
+
+#ifdef _MSC_VER
+#include <sys/stat.h>
+#define S_IXUSR 0
+#define S_IWUSR _S_IWRITE
+#define S_IRUSR _S_IREAD
+
+#define S_IXGRP S_IXUSR
+#define S_IWGRP S_IWUSR
+#define S_IRGRP S_IRUSR
+#define S_IXOTH S_IXUSR
+#define S_IWOTH S_IWUSR
+#define S_IROTH S_IRUSR
+#endif
+
+
+#ifdef _MSC_VER
+
+#ifndef __STDC__
+#define __STDC__ 1
+#include <io.h>
+#undef __STDC__
+#else
+#include <io.h>
+#endif
+
+#include <assert.h>
+#include <direct.h>
+#include <intrin.h>
+#include <malloc.h>
+#include <math.h>
+#include <signal.h>
+#include <stdarg.h>
+#include <stdio.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <string.h>
+#include <time.h>
+#include <Windows.h>
+#include <WinSock2.h>
+
+#include <pthread.h>
+#include <pthread_clock.h>
+
+
+// Do nothing for the system log for now.
+inline void openlog(const char*, int, int) {}
+inline void closelog() {}
+inline void syslog(int, const char*, ...) {}
+
+#define SIGCHLD 0
+// We use SIGILL as SIGALRM, as SIGILL is
+// never generated by windows.
+#define SIGALRM SIGILL
+// All 4 of these are invalid signal IDs.
+#define SIGVTALRM 26
+#define SIGPROF 27
+#define SIGKILL 28
+#define SIGUSR1 29
+
+inline void __stdcall __alarm_callback_func(HWND, UINT, UINT_PTR, DWORD) {
+  raise(SIGALRM);
+}
+
+inline unsigned int alarm(unsigned int seconds) {
+  SetTimer(NULL, 0, (UINT)(seconds * 1000), __alarm_callback_func);
+  return 0;
+}
+
+inline char* asctime_r(const tm* tm, char* buf) {
+  char tmpBuf[64];
+  if (asctime_s(tmpBuf, sizeof(tmpBuf), tm))
+    return NULL;
+  return strcpy(buf, tmpBuf);
+}
+
+inline void bzero(void* s, size_t n) {
+  ZeroMemory(s, n);
+}
+
+#define DT_UNKNOWN 0
+#define DT_DIR     1
+#define DT_REG     2
+#define DT_LNK     3
+struct dirent {
+  unsigned char d_type;
+  char* d_name;
+};
+
+struct DIR {
+  dirent dir;
+  HANDLE hand;
+  int cnt;
+  char name[MAX_PATH * 3];
+};
+
+inline int closedir(DIR* dir) {
```

Comment:
`FindClose` is indeed what I want, as this is closing the find handle, not a normal handle.

This returns an int because the standard `closedir` API returns an int. I've added a check for errors on the `FindClose` call, but I won't be marking the function for analysis, as this is intended as a functional portability mechanism, not an extensive API.